.github/workflows/yapf.yml

```diff
@@ -0,0 +1,12 @@
+name: YAPF Formatting Check
+on: [push]
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: run YAPF to test if python code is correctly formatted
+      uses: AlexanderMelde/yapf-action@v1.0
+      with:
+        args: --verbose
```

Comment:
I am not sure about this action, does it run yapf in the main directory? If so, wouldn't it be better to specify the `ItsPrompt` directory by adding `./ItsPrompt/ --recursive` to the args?

Reply:
Here [are the docs](https://github.com/AlexanderMelde/yapf-action) 

The action's [entrypoint is using](https://github.com/AlexanderMelde/yapf-action/blob/master/entrypoint.sh#LL4C1-L4C53
) `--recursive $GITHUB_WORKSPACE`. $GITHUB_WORKSPACE [resolves to the default working directory ](https://docs.github.com/en/actions/learn-github-actions/variables) on the runner for steps, and the default location of your repository when using the [checkout](https://github.com/actions/checkout) action

Reply:
So it looks like there is no action needed, am I right?

Reply:
Well, I think that is done. Will now resolve this and merge it.